pyproject.toml

```
@@ -22,9 +22,9 @@ dependencies = [
     "Authlib ~=1.2.0",
     "pyjwt ~=2.8.0",
     "requests ~=2.31.0",
-    "typing-extensions ~=4.8.0",
+    "typing-extensions ~=4.11.0",
```

```suggestion
    "typing-extensions >=4.11.0",
```

In the [typing-extensions documentation they specify that a minimum major version can be specified since they are following semantic versioning and increment the version if there are backwards-incompatible changes](https://typing-extensions.readthedocs.io/en/latest/#versioning-and-backwards-compatibility):

>Starting with version 4.0.0, typing_extensions uses [Semantic Versioning](https://semver.org/). A changelog is maintained [on GitHub](https://github.com/python/typing_extensions/blob/main/CHANGELOG.md).
>
>The major version is incremented for all backwards-incompatible changes. Therefore, it’s safe to depend on typing_extensions like this: typing_extensions >=x.y, <(x+1), where x.y is the first version that includes all features you need. In view of the wide usage of typing_extensions across the ecosystem, we are highly hesitant to break backwards compatibility, and we do not expect to increase the major version number in the foreseeable future.

Other libraries are using `>=` instead of `~=` for the version constraint, here's an example from pydantic: https://github.com/pydantic/pydantic/blob/main/pyproject.toml#L49

When I ran into issue https://github.com/kinde-oss/kinde-python-sdk/issues/39 there was a list of dependencies and their typing-extensions version constraint and they all used `>=`.

---

Hi @rudolfolah, thanks for pointing this out, but this pull request is a bit out of date in light of #41. I would like a second opinion from @rairaman before incorporating this in a new pull request if thats ok.

---

sounds good to me and glad to see the version was upgraded in the other PR.

my suggested change should also include the major version constraint:

```suggestion
    "typing-extensions >=4.11.0, <5",
```